components/data_comps/datm/bld/build-namelist

```diff
@@ -381,9 +381,7 @@ if ( $GRID eq "CLM_USRDAT" && $CLM_USRDAT_NAME =~ /^(UNSET|)$/ ) {
 EOF
 }
 
-(-d $DIN_LOC_ROOT)  or  die <<"EOF";
-** $ProgName - CESM inputdata root is not a directory: \"$DIN_LOC_ROOT\" **
-EOF
+(-d $DIN_LOC_ROOT)  or mkdir $DIN_LOC_ROOT;
```

Comment:
So, for example, it should be done in the setup script, or in the build script, rather than in build-namelist, correct?

I'd be fine with either of those, in which case you're right this should error, since build-namelist called by things that should have created the directory.